### DISCUSSION TITLE
Next.js api support?

### DISCUSSION THREAD

**Discussion by pixelkoduje** [apify/got-scraping#143] [Q&A]
Not working with next.js 14 api(app folder)

**Comment by pixelkoduje** [comment]
Any help?

**Reply by B4nan** [reply]
What problem do you actually have? "not working" is not a reproduction, we can't help with that really.

Note that those issues are usually about bundling and often have workarounds in the config of the underlying bundler.

**Comment by Hamza5** [comment]
I have the same bundling problem of Next.js, and the standard solution for those packages is to add them to `serverComponentsExternalPackages` in `next.config.mjs`, like this:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
    experimental: {
        serverComponentsExternalPackages: [
            'playwright', 'playwright-core', 'playwright-extra', 'puppeteer-extra-plugin-stealth', 'got-scraping'
        ]
    }
};

export default nextConfig;
```
The original package `got` works without this workaround. This is only needed for `got-scraping`.

**Reply by rdadoune** [reply]
I just had a similar issue. I was running Apollo via `/api/graphql` just fine, and the moment I included `got-scraper`, the endpoint started returning 404. Unfortunately, marking `got-scraping` as external gave me a new error:  `ERR_PACKAGE_PATH_NOT_EXPORTED`. Digging further for the root cause, I was able to determine that the real problem is a dependency package called `header-generator` because it includes some static JSON files that don't bundle nicely. I was able to resolve the issue by adding `header-generator` to [serverExternalPackages](https://nextjs.org/docs/app/api-reference/next-config-js/serverExternalPackages) (NextJS v15) in my `next.config.js` file.

**Reply by Hamza5** [reply]
> I just had a similar issue. I was running Apollo via `/api/graphql` just fine, and the moment I included `got-scraper`, the endpoint started returning 404. Unfortunately, marking `got-scraping` as external gave me a new error: `ERR_PACKAGE_PATH_NOT_EXPORTED`. Digging further for the root cause, I was able to determine that the real problem is a dependency package called `header-generator` because it includes some static JSON files that don't bundle nicely. I was able to resolve the issue by adding `header-generator` to [serverExternalPackages](https://nextjs.org/docs/app/api-reference/next-config-js/serverExternalPackages) (NextJS v15) in my `next.config.js` file.

Thanks for the information. I am still on Next.js 14 and haven't used 15 yet. It seems that `serverExternalPackages` is the new name for `serverComponentsExternalPackages`.
